File: TransCoder_tokenized_test_set_functions/python/GIVEN_SORTED_ARRAY_NUMBER_X_FIND_PAIR_ARRAY_WHOSE_SUM_CLOSEST_X.py
def printClosest ( arr, n, x ) :
    res_l, res_r = 0, 0
    l, r, diff = 0, n - 1, MAX_VAL
    while r > l :
        if abs ( arr [ l ] + arr [ r ] - x ) < diff :
            res_l = l
            res_r = r
            diff = abs ( arr [ l ] + arr [ r ] - x )
        if arr [ l ] + arr [ r ] > x :
            r -= 1
        else :
            l += 1
    print ( 'The closest pair is {} and {}' . format ( arr [ res_l ], arr [ res_r ] ) )

File: TransCoder_tokenized_test_set_functions/java/GIVEN_SORTED_ARRAY_NUMBER_X_FIND_PAIR_ARRAY_WHOSE_SUM_CLOSEST_X.java
static void printClosest ( int arr [ ], int n, int x ) {
  int res_l = 0, res_r = 0;
  int l = 0, r = n - 1, diff = Integer . MAX_VALUE;
  while ( r > l ) {
    if ( Math . abs ( arr [ l ] + arr [ r ] - x ) < diff ) {
      res_l = l;
      res_r = r;
      diff = Math . abs ( arr [ l ] + arr [ r ] - x );
    }
    if ( arr [ l ] + arr [ r ] > x ) r --;
    else l ++;
  }
  System . out . println ( " The closest pair is " + arr [ res_l ] + " and " + arr [ res_r ] );
}